jctools-core/src/main/java/org/jctools/queues/atomic/BaseMpscLinkedAtomicArrayQueue.java

```diff
@@ -126,78 +234,35 @@ public boolean offer(final E e) {
                 continue;
             }
             // pIndex is even (lower bit is 0) -> actual index is (pIndex >> 1)
-
             // mask/buffer may get changed by resizing -> only use for array access after successful CAS.
             mask = this.producerMask;
             buffer = this.producerBuffer;
-            // a successful CAS ties the ordering, lv(pIndex)-[mask/buffer]->cas(pIndex)
-
             // assumption behind this optimization is that queue is almost always empty or near empty
             if (producerLimit <= pIndex) {
                 int result = offerSlowPath(mask, pIndex, producerLimit);
-                switch (result) {
-                case 0:
-                    break;
-                case 1:
-                    continue;
-                case 2:
-                    return false;
-                case 3:
-                    resize(mask, buffer, pIndex, e);
-                    return true;
+                switch(result) {
+                    case 0:
+                        break;
+                    case 1:
+                        continue;
+                    case 2:
+                        return false;
+                    case 3:
+                        resize(mask, buffer, pIndex, e);
+                        return true;
                 }
             }
-
             if (casProducerIndex(pIndex, pIndex + 2)) {
                 break;
             }
         }
         // INDEX visible before ELEMENT, consistent with consumer expectation
         final int offset = modifiedCalcElementOffset(pIndex, mask);
-        buffer.lazySet(offset, e);
```

Comment:
Swapped this to using the soElement method in the util classes for easy transformation by the generator